chainer_chemistry/models/ggnn_gwm.py

```diff
@@ -0,0 +1,133 @@
+                 dropout_ratio=0.5,
+                 concat_hidden=False,
+                 weight_tying=True, activation=functions.identity,
+                 num_edge_type=4):
+        super(GGNN_GWM, self).__init__()
+        n_readout_layer = n_layers if concat_hidden else 1
+        n_message_layer = 1 if weight_tying else n_layers
+        with self.init_scope():
+            # Update
+            self.embed = EmbedAtomID(out_size=hidden_dim, in_size=n_atom_types)
+            self.update_layers = chainer.ChainList(*[GGNNUpdate(
+                hidden_dim=hidden_dim, num_edge_type=num_edge_type)
+                for _ in range(n_message_layer)])
+            # GWM
+            self.embed_super = links.Linear(in_size=n_super_feature, out_size=hidden_dim_super)
+            self.gwm = GWM(hidden_dim=hidden_dim, hidden_dim_super=hidden_dim_super,
+                 n_layers=n_message_layer, n_heads=n_heads,
+                 dropout_ratio=dropout_ratio,
+                 tying_flag=weight_tying,
+                 gpu=-1)
+            # Readout
+            self.readout_layers = chainer.ChainList(*[GGNNReadout(
+                out_dim=out_dim, hidden_dim=hidden_dim,
+                activation=activation, activation_agg=activation)
+                for _ in range(n_readout_layer)])
+            self.linear_for_concat_super = links.Linear(in_size=None, out_size=out_dim)
+        self.out_dim = out_dim
+        self.hidden_dim = hidden_dim
+        self.hidden_dim_super = hidden_dim_super
+        self.n_layers = n_layers
+        self.n_heads = n_heads
+        self.dropout_ratio = dropout_ratio
+        self.num_edge_type = num_edge_type
+        self.activation = activation
+        self.concat_hidden = concat_hidden
+        self.weight_tying = weight_tying
+
+    def __call__(self, atom_array, adj, super_node, is_real_node=None):
+        """Forward propagation
+
+        Args:
+            atom_array (numpy.ndarray): minibatch of molecular which is
+                represented with atom IDs (representing C, O, S, ...)
+                `atom_array[mol_index, atom_index]` represents `mol_index`-th
+                molecule's `atom_index`-th atomic number
+            adj (numpy.ndarray): minibatch of adjancency matrix with edge-type
+                information
+            super_node (numpy.ndarray): 1D rray, the super-node observation.
+            is_real_node (numpy.ndarray): 2-dim array (minibatch, num_nodes).
+                1 for real node, 0 for virtual node.
+                If `None`, all node is considered as real node.
+
+        Returns:
+            ~chainer.Variable: minibatch of fingerprint
+        """
+        # reset state
+        self.reset_state()
+        if atom_array.dtype == self.xp.int32:
+            h = self.embed(atom_array)  # (minibatch, max_num_atoms)
+        else:
+            h = atom_array
+        h0 = functions.copy(h, cuda.get_device_from_array(h.data).id)
+
+        self.gwm.GRU_local.reset_state()
```

Comment:
I think we can include this inside `reset_state` method.

Reply:
fixed